fdb-extensions/src/main/java/com/apple/foundationdb/half/Half.java

```diff
@@ -0,0 +1,812 @@
+     * format" bit layout.
+     *
+     * <p>
+     * Bit 15 (the bit that is selected by the mask {@code 0x8000}) represents the sign of the floating-point number.
+     * Bits 14-10 (the bits that are selected by the mask {@code 0x7c00}) represent the exponent. Bits 9-0 (the bits
+     * that are selected by the mask {@code 0x03ff}) represent the significand (sometimes called the mantissa) of the
+     * floating-point number.
+     *
+     * <p>
+     * If the argument is positive infinity, the result is {@code 0x7c00}.
+     *
+     * <p>
+     * If the argument is negative infinity, the result is {@code 0xfc00}.
+     *
+     * <p>
+     * If the argument is NaN, the result is {@code 0x7e00}.
+     *
+     * <p>
+     * In all cases, the result is a short that, when given to the {@link #shortBitsToHalf(short)} method, will produce
+     * a floating-point value the same as the argument to {@code halfToShortBits} (except all NaN values are collapsed
+     * to a single "canonical" NaN value).
+     *
+     * @param half
+     *            a Half object.
+     *
+     * @return the bits that represent the floating-point number.
+     */
+    public static short halfToShortBits(Half half) {
+        if (!isNaN(half)) {
+            return halfToRawShortBits(half);
+        }
+        return 0x7e00;
+    }
+
+    /**
+     * Returns a representation of the specified floating-point value according to the IEEE 754 floating-point "single
+     * format" bit layout, preserving Not-a-Number (NaN) values.
+     *
+     * <p>
+     * Bit 15 (the bit that is selected by the mask {@code 0x8000}) represents the sign of the floating-point number.
+     * Bits 14-10 (the bits that are selected by the mask {@code 0x7c00}) represent the exponent. Bits 9-0 (the bits
+     * that are selected by the mask {@code 0x03ff}) represent the significand (sometimes called the mantissa) of the
+     * floating-point number.
+     *
+     * <p>
+     * If the argument is positive infinity, the result is {@code 0x7c00}.
+     *
+     * <p>
+     * If the argument is negative infinity, the result is {@code 0xfc00}.
+     *
+     * <p>
+     * If the argument is NaN, the result is the integer representing the actual NaN value. Unlike the
+     * {@code halfToShortBits} method, {@code halfToRawShortBits} does not collapse all the bit patterns encoding a NaN
+     * to a single "canonical" NaN value.
+     *
+     * <p>
+     * In all cases, the result is a short that, when given to the {@link #shortBitsToHalf(short)} method, will produce
+     * a floating-point value the same as the argument to {@code halfToRawShortBits}.
+     *
+     * @param half
+     *            a Half object.
+     *
+     * @return the bits that represent the half-point number.
+     */
+    public static short halfToRawShortBits(Half half) {
+        return floatToHalfShortBits(half.floatRepresentation);
+    }
+
+    private static short floatToHalfShortBits(float floatValue) {
```

Comment:
I wonder if this method should be made public. It would allow, for example, the `HalfVector` encoding class to avoid creating a `Half` object for each dimension, and instead, it would be able to go directly from `double` (or `float`) to `short` directly.